README.md

```diff
@@ -11,11 +13,172 @@ The modern, distributed and scalable implementation of a game inspired by Connec
+We decided to use the cloud provider [Linode](https://www.linode.com/) because they provide fair pricing and a good free initial credit.
+
+Setting up hosting was very easy, you just copy the docker-compose.yml file, replace the placeholder jwt-secret with a generated one and you're almost ready to start.
+If you don't already have a valid certificate around, you can easily generate one using [Certbot](https://certbot.eff.org/) using a command similar to the following and mounting them to their corresponding location within the Speedwagon container:
+
+`sudo docker run -it --rm --name certbot -v "/etc/letsencrypt:/etc/letsencrypt" -v "/var/lib/letsencrypt:/var/lib/letsencrypt" -p 80:80 certbot/certbot certonly --standalone`
+
+# Design Decisions
+
+## gRPC
+We wanted real time updates and gRPC provides a way to stream real time updates.
+Implementing real time update from server-to-server and server-to-browser is no easy task and implementing a type safe variant through websockets would have taken a lot longer than just using a gRPC library.
+
+## Micronaut
+
+Micronaut offers a couple of benefits:
+
+- It supports gRPC out-of-the-box
+- It offers Configuration Management
+- It provides fast startup and performance by using build-time dependency injection
+
+## Kotlin
+
+Kotlin is a modern, concise, and `null` safe programming language, which offer great language and ecosystem support for gRPC.
+For example, implementing parallel code using Kotlin coroutines is vastly easier than using plain old java.
+
+## Svelte
+
+Svelte is a modern, light-weight SPA framework that meets the needs of small projects like ours.
+
+## Envoy
+
+Envoy is the only reverse proxy with official and native `grpc-web` support known to us.
+
+Envoy uses round-robin for load balancing and only considers servers for load balancing which fulfill a readiness check.
+For example, a Joestar instance without a valid rohan connection is not considered ready and will not be load-balanced.
+
+## Symmetric JWT
+
+We couldn't easily use the `micronaut-security` package because the feature is still WIP for gRPC (see [`micronaut-grpc` issue #164](https://github.com/micronaut-projects/micronaut-grpc/issues/164)).
+The official token-based authentication works by using Google as a token provider, but we didn't want to have a vendor lock-in.
+
+In the end, we decided to use simple symmetric tokens because of the project scope.
+
+The client does use the metadata to send the token [which was recommended back in 2018 by the `grpc-web` team](https://github.com/grpc/grpc-web/issues/207#issuecomment-406134504).
+
+## Notifications
+
+In order to preserve network bandwidth, which can be pricey depending on the hosting provider, we use real-time updates with light-weight change events instead of complete states.
+
+# Operation
 The easiest way to get ChaosConnect running is using docker-compose. We do not provider support for running the software components otherwise.
 
 ## Prerequisites
```

Comment:
Maybe also document that for local development you need to have JDK16, Node 16 and a modern version of docker installed. Our project also contains run configurations for IntelliJ, which is our IDE of choice.

Reply:
Done